src/driver/sqljs/SqljsDriver.ts

```diff
@@ -199,9 +199,10 @@ export class SqljsDriver extends AbstractSqliteDriver {
      * If a custom autoSaveCallback is specified, it get's called with the database as Uint8Array,
      * otherwise the save method is called which saves it to file (Node.js), local storage (browser)
      * or indexedDB (browser with enabled useLocalForage option).
+     * Don't auto-save when in transaction as the call to export will end the current transaction
      */
     async autoSave() {
-        if (this.options.autoSave) {
+        if (this.options.autoSave && !this.queryRunner?.isTransactionActive) {
```

Comment:
shouldn't we throw an error if one tries to autoSave during active transaction? Imagine someone called autoSave but it didn't save anything if transaction was active, and user still thinks everything was saved.

Reply:
I get your point though it would be a bit tricky to do it as this method is automatically called on query runner release if database has changed (code from `SqljsQueryRunner.ts`):
```ts
private async flush() {
    if (this.isDirty) {
            await this.driver.autoSave()
            this.isDirty = false
    }
}

async release(): Promise<void> {
    await this.flush()
    return super.release()
}
```

If we throw an error everytime this gets called within a transaction, we will end up with tons of errors as a simple `update` on the repository during a transaction will trigger it. Is there a way to inform the use without throwing an error?

Reply:
okay, another question:

shouldn't we add the proposed check into `commitTransaction` override?:

```ts
    async commitTransaction(): Promise<void> {
        await super.commitTransaction()
        if (this.isTransactionActive === false)
            await this.flush()
    }
```

Reply:
You want to add it there to ensure that the database is saved after the transaction?

Currently, `COMMIT` sets `isDirty` to `true`, then, when the query runner is released after the query, it automatically flushes the database as there is no active transaction.

So, database is already saved after transaction but it would be easier to understand I admit.

Reply:
It makes sense to have it there I think. We flush after the transaction. If do it only if transaction was committed.